lisa/trace.py

```diff
@@ -294,7 +314,812 @@ def get_view(self, window):
+
+            # Write the Parquet file and update the write speed
+            with measure_time() as measure:
+                if self.DATAFRAME_SWAP_FORMAT == 'parquet':
+                    # Snappy compression seems very fast
+                    data.to_parquet(df_path, compression='snappy')
+                else:
+                    raise ValueError('Dataframe swap format "{}" not handled'.format(self.DATAFRAME_SWAP_FORMAT))
+
+            # Update the swap
+            swap_entry_path = os.path.join(self.swap_dir, swap_entry.meta_filename)
+            swap_entry.to_path(swap_entry_path)
+            self._swap_content[swap_entry.pd_desc_nf] = swap_entry
+
+            # Assume that reading from the swap will take as much time as
+            # writing to it. We cannot do better anyway, but that should
+            # mostly bias to keeping things in memory if possible.
+            swap_cost = measure.exclusive_delta
+            data_swapped_size = os.stat(df_path).st_size
+
+            mem_usage = self._data_mem_usage(data)
+            if mem_usage:
+                self._update_swap_cost(swap_cost / mem_usage)
+            self._swap_size += data_swapped_size
+            self._update_data_swap_size_estimation(data, data_swapped_size)
+            self.scrub_swap()
+
+    def _get_swap_size(self):
+        if self.swap_dir:
+            return sum(
+                dir_entry.stat().st_size
+                for dir_entry in os.scandir(self.swap_dir)
+            )
+        else:
+            return 0
+
+    def scrub_swap(self):
+        """
+        Scrub the swap area to remove old files if the storage size limit is exceeded.
+        """
+        if self._swap_size > self.max_swap_size and self.swap_dir:
+            stats = {
+                dir_entry.name: dir_entry.stat()
+                for dir_entry in os.scandir(self.swap_dir)
+            }
+
+            data_files = {
+                swap_entry.data_filename: swap_entry
+                for swap_entry in self._swap_content.values()
+            }
+
+            # Get rid of stale files that are not referenced by any swap entry
+            metadata_files = {
+                swap_entry.meta_filename
+                for swap_entry in self._swap_content.values()
+            }
+            metadata_files.add(self.TRACE_META_FILENAME)
+            non_stale_files = data_files.keys() | metadata_files
+            stale_files = stats.keys() - non_stale_files
+            for filename in stale_files:
+                del stats[filename]
+                path = os.path.join(self.swap_dir, filename)
+                os.unlink(path)
+
+            def by_mtime(path_stat):
+                path, stat = path_stat
+                return stat.st_mtime
+
+            # Sort by modification time, so we discard the oldest caches
+            total_size = 0
+            discarded_swap_entries = set()
+            for filename, stat in sorted(stats.items(), key=by_mtime):
```

Comment:
Do you think it's worth maintaining extra state for that ? _scrub_swap() needs to work with a swap area full of junk anyway (from previous implem). It's also best if it can reasonably easily be adapted to a multithread/process setup, where the same swap area could be (unknowingly) shared.

Reply:
It's an implementation detail that wouldn't have any visible impact other than time taken to `scrub_swap()`. It's a good way to implement LRU caches, so if we notice `scrub_swap()` is a bit lengthy this would be one area to improve. Not a big stopper for me, but maybe could be worth a `# TODO`